server/server.py

```diff
@@ -180,6 +181,78 @@ def count_verified():
     return jsonify({'verified_count': count})
 
 
+@app.route('/submit/testimonial', methods=['POST'])
+@schema.validate({
+    'properties': {
+        'name': {'type': 'string'},
+        'student_number': {'type': 'number'},
+        'program': {'type': 'string'},
+        'year': {'type': 'string'},
+        'testimonial': {'type': 'string', 'maxLength': 1500},
+        'anonymous': {'type': 'boolean'}
+    },
+    'required': ['name', 'student_number', 'program', 'year', 'testimonial']
+})
+def submit_testimonial():
+    '''
+    Submit a testimonial to be approved by the team
+    '''
+    logger.info('Hit route %s', request.path)
+
+    data = request.get_json()
+
+    logger.debug(data)
+
+    bytes_student_number = int_to_bytes(data['student_number'])
+    bytes_student_number = data['name'].encode('utf-8')
+
+    name = ''
+
+    if data['anonymous'] is True:
+        name = 'Anonymous'
+    else:
+        name = data['name'].capitalize().split(' ')[0]

```

Comment:
Some people may object to their name being capitalized, but yeah xD
I had a prof who had a question mark as a quote-unquote real part of his name.

Also this (nor the json schema above) checks for blank names; I assume that's okay? Thought it would be better to mention it and double check

Reply:
the methods wouldn't cause an error, and if the person leaves the name blank it won't matter since it likely won't match an entry in the petition signer's collection

Reply:
Fair enough! I know it won't cause an error, I just wanted to check if there was other validation or that was okay with you